src/data/transform/mod.rs

```diff
@@ -35,7 +35,13 @@ pub fn deserialize_row(
                 let raw_value = row
                     .get(column_index)
                     .context("failed to get column of row")?;
-                let json_value = get_json_value_from_string(raw_value);
+
+                let json_value = match get_json_value_from_string(raw_value, &path_mapping.column_type) {
+                    Ok(value) => { value }
+                    Err(error) => {
+                        return Err(anyhow::anyhow!("column \"{}\": {error}", headers[column_index].to_string()));
+                    }
+                };
```

Comment:
There is also this `Context` trait in the `anyhow` crate, see: https://docs.rs/anyhow/latest/anyhow/index.html

That should allow you to more easily add "context" to errors, something like:
```suggestion
                let json_value = match get_json_value_from_string(raw_value, &path_mapping.column_type).with_context(|| format!("error in column \"{}\"", headers[column_index].to_string()))?;
```
Note: in this case `with_context` is better, because it is evaluate lazyly and formatting a string isn't zero cost. If you simply need a static string slice, the normal `context` method is just fine.

Reply:
That is very useful.  
I'll look into that.